lib/jormungandr/src/Cardano/Wallet/Jormungandr/Transaction.hs

```diff
@@ -1,19 +1,100 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
 
 module Cardano.Wallet.Jormungandr.Transaction
     ( newTransactionLayer
     ) where
 
 import Prelude
 
+import Cardano.Wallet.Jormungandr.Compatibility
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( Depth (AddressK), Key, Passphrase (..), XPrv, getKey )
+import Cardano.Wallet.Primitive.CoinSelection
+    ( CoinSelection (..) )
+import Cardano.Wallet.Primitive.Types
+    ( Hash (..), Tx (..), TxId (..), TxOut (..), TxWitness (..) )
 import Cardano.Wallet.Transaction
-    ( TransactionLayer (..) )
+    ( ErrMkStdTx (..), TransactionLayer (..) )
+import Control.Monad
+    ( forM )
+import Data.ByteString
+    ( ByteString )
+import Data.Quantity
+    ( Quantity (..) )
 
+import qualified Cardano.Crypto.Wallet as CC
 
 -- | Construct a 'TransactionLayer' compatible with Shelley and 'Jörmungandr'
-newTransactionLayer :: TransactionLayer t
-newTransactionLayer = TransactionLayer
-    { mkStdTx = error "TODO: See http-bridge as starting point"
-    , estimateSize = error "TODO: See http-bridge as starting point"
+newTransactionLayer
+    :: forall n . Hash "Block0Hash"
+    -> TransactionLayer (Jormungandr n)
+newTransactionLayer block0Hash = TransactionLayer
+    { mkStdTx = \keyFrom inps outs -> do
+        let ins = (fmap fst inps)
+        let tx = Tx ins outs
+        let witData = witnessUtxoData block0Hash (txId @(Jormungandr n) tx)
+        txWitnesses <- forM inps $ \(_in, TxOut addr _c) -> mkWitness witData
+            <$> withEither (ErrKeyNotFoundForAddress addr) (keyFrom addr)
+        return (tx, txWitnesses)
+    , estimateSize = \(CoinSelection inps outs chngs) ->
```

Comment:
This can probably be left out for know by the way and implemented as `const (Quantity 0)`. Jormungandr doesn't support fee anyway at the moment...